model/AbstractRdsResultStorage.php

```diff
@@ -0,0 +1,727 @@
+    }
+
+    /**
+     * Prepares data to be inserted in database according to a given schema.
+     *
+     * @param string   $deliveryResultIdentifier
+     * @param string   $test
+     * @param Variable $variable
+     * @param string   $callId
+     *
+     * @return array
+     */
+    abstract protected function prepareVariableDataForSchema($deliveryResultIdentifier, $test, Variable $variable, $callId);
+
+    /**
+     * Builds a variable from database row.
+     *
+     * @param array $variable
+     *
+     * @return \stdClass
+     */
+    protected function getResultRow($variable)
+    {
+        $resultVariable = $this->unserializeVariableValue($variable[self::VARIABLE_VALUE]);
+        $object = new \stdClass();
+        $object->uri = $variable[self::VARIABLES_TABLE_ID];
+        $object->class = get_class($resultVariable);
+        $object->deliveryResultIdentifier = $variable[self::VARIABLES_FK_COLUMN];
+        $object->callIdItem = $variable[self::CALL_ID_ITEM_COLUMN];
+        $object->callIdTest = $variable[self::CALL_ID_TEST_COLUMN];
+        $object->test = $variable[self::TEST_COLUMN];
+        $object->item = $variable[self::ITEM_COLUMN];
+        $object->variable = clone $resultVariable;
+
+        return $object;
+    }
+
+    /**
+     * @return QueryBuilder
+     */
+    protected function getQueryBuilder()
+    {
+        return $this->getPersistence()->getPlatform()->getQueryBuilder();
+    }
+
+    /**
+     * @return \common_persistence_SqlPersistence
+     */
+    public function getPersistence()
+    {
+        if ($this->persistence === null) {
+            $persistenceId = $this->hasOption(self::OPTION_PERSISTENCE) ?
+                $this->getOption(self::OPTION_PERSISTENCE)
+                : 'default';
+            $this->persistence = $this->getServiceLocator()->get(PersistenceManager::SERVICE_ID)->getPersistenceById($persistenceId);
+        }
+
+        return $this->persistence;
+    }
+
+    /**
+     * @param $value
+     *
+     * @return mixed
+     */
+    protected function unserializeVariableValue($value)
+    {
+        return unserialize($value);
+    }
+
+    /**
+     * @param $value
+     *
+     * @return string
+     */
+    protected function serializeVariableValue($value)
+    {
+        return serialize($value);
+    }
+
+    /**
+     * @param array $data
+     *
+     * @throws DuplicateVariableException
+     */
+    private function insertMultiple(array $data)
+    {
+        $duplicatedData = false;
+        try {
+            $this->getPersistence()->insertMultiple(self::VARIABLES_TABLENAME, $data);
+        } catch (UniqueConstraintViolationException $e) {
+            $duplicatedData = true;
+            foreach ($data as $row) {
+                try {
+                    $this->getPersistence()->insert(self::VARIABLES_TABLENAME, $row);
+                } catch (UniqueConstraintViolationException $e) {
+                    //do nothing, just skip it
```

Comment:
do we really need this code block ?

Reply:
No idea. All this class is not new code but renamed file.